apps/lead/models.py

```diff
@@ -218,6 +223,29 @@ def get_for(cls, user):
             # Or, return nothing if view_all is not present
             models.Q(view_all=view_perm_value)
         )
+        # filter entries
+        entries_filter_data = filters.get('entries_filter_data', {})
+        qs = qs.annotate(
+            filtered_entries_count=models.functions.Coalesce(
+                models.Subquery(
+                    get_filtered_entries(user, entries_filter_data).filter(
+                        lead=models.OuterRef('pk')
+                    ).annotate(
+                        count=models.Count('id')
+                    ).values('count')[:1], output_field=models.IntegerField()
+                ), 0
+            ),
+            verified_filtered_entries_count=models.functions.Coalesce(
+                models.Subquery(
+                    get_filtered_entries(user, entries_filter_data).filter(
+                        lead=models.OuterRef('pk'),
+                        verified=True
+                    ).annotate(
+                        count=models.Count('id')
+                    ).values('count')[:1], output_field=models.IntegerField()
+                ), 0
+            ),
+        )
```

Comment:
If `entries_filter_data` is empty, Can we use the `no_of_entries` which is calculated below?
Add this annotate after the current annotation.

Also, let's change `no_of_entries` to `entries_count`. cc: @AdityaKhatri 

Something like this?
```suggestion
        entries_filter_data = filters.get('entries_filter_data', {})
        return qs.annotate(
            no_of_entries=models.Count('entry', distinct=True),
            assessment_id=models.F('assessment'),
            verified_entries_count=models.Count('entry', filter=models.Q(entry__verified=True)),

            filtered_entries_count=(
                models.functions.Coalesce(
                    models.Subquery(
                        get_filtered_entries(user, entries_filter_data).filter(
                            lead=models.OuterRef('pk')
                        ).annotate(
                            count=models.Count('id')
                        ).values('count')[:1], output_field=models.IntegerField()
                    ), 0
                ) if entries_filter_data else F('no_of_entries')
            ),
            filtered_verified_entries_count=(
                models.functions.Coalesce(
                    models.Subquery(
                        get_filtered_entries(user, entries_filter_data).filter(
                            lead=models.OuterRef('pk'),
                            verified=True
                        ).annotate(
                            count=models.Count('id')
                        ).values('count')[:1], output_field=models.IntegerField()
                    ), 0
                ) if entries_filter_data else F('verified_entries_count')
            )
        )
```

Reply:
Okay, lets make it consistent